### ISSUE TITLE
Update builtins-list.txt

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#3487]
Imported from SourceForge on 2024-07-08 01:55:59
Created by **[aaroncl](https://sourceforge.net/u/aaroncl/)** on 2020-09-18 08:26:35
Original: https://sourceforge.net/p/maxima/bugs/3656

---

Running a shell command like this one:
grep -Eroh -e '^\(defmvar \$\w+' -e '^\(defmfun \$\w+' src/ | awk '{print $2}' | tr -d '$' | sort | uniq | comm -13 <(sort share/builtins-list.txt) - | less
I get a list of maxima function/variable name (defined with defmfun and defmvar) not included in builtins-list.txt. I can not determine which ones are important enough to be included in the file. (see attachment)
But I am sure those that are documented in the manual and can be described with `? keyword` should be included. (e.g. encode_time, decode_time, firstn, lastn). 

p.s. And I notice the function eulerpoly (similar to bernpoly) is not documented in the manual.

Attachments:
- https://sourceforge.net/p/maxima/bugs/3656/attachment/diff.txt

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 01:56:00
Created by **[aaroncl](https://sourceforge.net/u/aaroncl/)** on 2020-09-18 08:27:31
Original: https://sourceforge.net/p/maxima/bugs/3656/#459e

---

A patch is supplied in my own view.



Attachments:
- https://sourceforge.net/p/maxima/bugs/_discuss/thread/3d13bd32a6/459e/attachment/builtins-list.txt.patch

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 01:56:04
Created by **[aaroncl](https://sourceforge.net/u/aaroncl/)** on 2020-09-18 08:30:52
Original: https://sourceforge.net/p/maxima/bugs/3656/#8403

---

Ah, the shell command should be:
`grep -Eroh -e '^\(defmvar \$\w+' -e '^\(defmfun \$\w+' src/ | awk '{print $2}' | tr -d '$' | sort | uniq | comm -13 <(sort share/builtins-list.txt) -`

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 01:56:08
Created by **[villate](https://sourceforge.net/u/villate/)** on 2020-09-18 09:50:26
Original: https://sourceforge.net/p/maxima/bugs/3656/#6da1

---

Hello,
Thanks for your contribution. I would modify your command in the following way, to include also commands which are in share, but are loaded automatically when needed:
~~~
grep -Eroh -e '^\(defmvar \$\w+' -e '^\(defmfun \$\w+' -e 'automexp \$w+' src/ | awk '{print $2}' | tr -d '$' | sort |sort <(sort share/builtins-list.txt) - |uniq
~~~
We now must decide whether if makes sense to add all those commands, or only the ones that are documented in the manual. I would be incline to add them all, regardless of being documented or not.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 01:56:11
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-06-30 06:58:26
Original: https://sourceforge.net/p/maxima/bugs/3656/#1642

---

Fixed by commit 8ecb2b1a1. I ran the command stated above by Jaime V. and committed the result.
